### DISCUSSION TITLE
how to select a sd-model when use text2image api?

### DISCUSSION THREAD

**Discussion by shuaiyy** [Meonako/webui-api#6] [Q&A] [answered]
I want to use a specific sd model when call the text2image api, how to to that?

I didn't find a field for  that, like`sd_model` or `ckpt_checksum` or `checkpoint` ...

**Comment by Lebenoa** [comment]
Use `OverrideSettings` and `DoNotOverrideSettingsRestoreAfterwards` to set this value

Settings will restore to it's previous state (before sending the API) if `DoNotOverrideSettingsRestoreAfterwards` is not set or set to `false`

![image](https://user-images.githubusercontent.com/76484203/233784032-42337d4f-9b0d-4e5a-bfeb-80d7588a258d.png)

As you can see in the image the common one is `sd_model_checkpoint`. You can find this (the keys) in Web UI settings by toggle something on and off ***OR*** browsing thru `config.json`. The `sd_model_checkpoint` you can find in `quick settings` field. 

If you want to I can make the field accept `api.Options` instead of `map[string]any`

P.S. Sorry for the informal and maybe rude answer. I'm not fluent in English

**Comment by Lebenoa** [comment, marked as answer]
I have update an [`example here`](https://github.com/Meonako/webui-api/wiki/Example#txt2img-with-overridesettings) or just view below

```go
package main

import (
    "fmt"
    "os"

    "pg/webui-api"
    "pg/webui-api/sampler"
    "pg/webui-api/upscaler"
)

func main() {
    // Currently using `realdosmix_.safetensors [0d27c62ffa]` which is a realistic model
    API := api.New(api.Config{
        Default: &api.Default{
            Steps:    28,
            Sampler:  sampler.DPM_PLUS_PLUS_2M_KARRAS,
            Width:    384,
            Height:   512,
            CFGScale: 7.0,
        },
    })

    response, err := API.Text2Image(&api.Txt2Image{
        Prompt: api.BuildPrompt(
            "(masterpiece), (high quality), highres, (high resolution illustration), bishoujo, (game cg), intricate details, detailed_face",
            "1girl, skinny, narrow waist, white pantyhose",
            "(white serafuku), (school uniform), (blue sailor collar), (red neckerchief), (pleated skirt)",
            "large breasts, lipgloss, hair bow, (absurdly_long_hair:1.5)",
            "beautiful sky and beautiful garden, cowboy_shot",
        ),
        NegativePrompt:    "(KHFB, AuroraNegative), EasyNegative, badhandv4, (worst Quality, low quality:1.4), (multiple_girls:1.2), border, nsfw, skimpy, grayscale, umbrella",
        EnableHR:          true,
        HRScale:           2,
        HRUpscaler:        upscaler.R_ESRGAN_4x_PLUS_ANIME_6B,
        DenoisingStrength: 0.5,
        OverrideSettings: map[string]any{
            "sd_model_checkpoint": "AbyssOrangeMix2_hard", // a anime, semi-realistic style model
        },
        DecodeAfterResult: true,
    })
    pif(err)

    fmt.Println("Successfully decoded", len(response.DecodedImages), "images")
    for idx, decoded := range response.DecodedImages {
        file, err := os.Create(fmt.Sprint("override result", idx, ".png"))
        pif(err)

        file.Write(decoded)
        file.Close()
    }
}

func pif(err error) {
    if err != nil {
        panic(err)
    }
}
```

**Reply by shuaiyy** [reply]
thanks! 

I'm try to build a group chatbot for AI drawing, and your SDK project can help me save a lot of time and coding.



> If you want to I can make the field accept `api.Options` instead of `map[string]any`

It's a good idea to add a field,  cause switching between different models while drawing is a very common scenario.


P.S.   I am not a native English speaker, and some of my expressions were translated using ChatGPT.
